system/security/shared/caching.cpp

```diff
@@ -686,6 +686,22 @@ void CPermissionsCache::managedFileScopesCacheFillThread()
         // completes after now is set which could result in elapsedTimeSinceLastFill being negative.
         if (elapsedTimeSinceLastFill >= m_cacheTimeoutInSeconds)
         {
+            try
+            {
+                fillManagedFileScopesCache(false);
+            }
+            catch (IException *e)
+            {
+                StringBuffer errorMsg;
+                e->errorMessage(errorMsg);
+                OERRLOG("CACHE: Exception retrieving managed scopes in fill thread, error=%d, message=%s", e->errorCode(), errorMsg.str());
+                e->Release();
+            }
+            catch (...)
+            {
+                OERRLOG("CACHE: Unknown exception retrieving managed scopes in fill thread");
+            }
+
             fillManagedFileScopesCache(false);
```

Comment:
as coded, this is now calling fillManagedFileScopesCache twice, once in the new try/catch block, and once outside.
Looks like you forgot to delete this line.

Reply:
Gavin also suggested that the code that clears the m_userDefaultFileScopePermissions cache shouldn't happen if fillManagedFileScopesCache fails.. but it continues to with this change.
I'm not sure it matters though, the code that queries m_userDefaultFileScopePermissions, always re-adds entries if missing.

Reply:
The problem is that if LDAP is down (likely if we have hit an error mesage) that will not be refilled.

Reply:
Went ahead and moved default cache clearing so that it does not happen if there is an exception retrieving the managed scope tree. Best case we keep cached default permissions a while longer. Worst case, the AD has failed and we are gonna experience that somewhere else anyway.